src/assets/Generator.Shared/RequestHeaderExtensions.cs

```diff
@@ -6,11 +6,28 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
+using System.Linq;
+using System.Xml;
 
 namespace Azure.Core
 {
     internal static class RequestHeaderExtensions
     {
+        internal static string ConvertToString(object? value, string? format = null)
+            => value switch
+            {
+                null => "null",
+                string s => s,
+                bool b => TypeFormatters.ToString(b),
+                int or float or double or long or decimal => ((IFormattable)value).ToString(TypeFormatters.DefaultNumberFormat, CultureInfo.InvariantCulture),
+                byte[] b when format != null => TypeFormatters.ToString(b, format),
+                IEnumerable<string> s => string.Join(",", s),
+                DateTimeOffset dateTime when format != null => TypeFormatters.ToString(dateTime, format),
+                TimeSpan timeSpan when format != null => TypeFormatters.ToString(timeSpan, format),
+                TimeSpan timeSpan => XmlConvert.ToString(timeSpan),
+                Guid guid => guid.ToString(),
+                _ => value.ToString()!
+            };
```

Comment:
Is this the same as a code snippet somewhere else? I remember I wrote similar things like this in the Query thing. We should not duplicate it

Reply:
I moved this function to `TypeFormatter` as a common one.